### PR TITLE
CADC-10685: update nimbus library for OIDC support...

### PR DESCRIPTION
**Pull request by hjeeves**
…update claims declaration so they are populated as expected. AuthorizeAction uses RegistryClient reg-applications lookup support to remove hard coded prod login page reference.

### CODE REVIEW

File: cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/OIDCUtil.java [hunk 1]
```diff
@@ -325,17 +326,22 @@ public static String buildIDToken(RelyParty rp, boolean isUserInfo) throws Excep
...
-                builder.claim("aud", rp.getClientID());
-            }
-            builder.claim("iat", calendar.getTime());
+
+            builder.setAudience(getClaimIssuer())   // aud
+                .setIssuedAt(calendar.getTime())    // iat
+                .setIssuer(clientID)                // iss
+                .setSubject(numericPrincipal.getName()) // sub
+                .setHeaderParam("typ", "JWT")
+                .setHeaderParam("alg", "RSA256");
```
Comment (brianmajor) [inline cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/OIDCUtil.java:339]: I'm still surprised that the 'audience' is us and not the rely party, and that the 'issuer' is rely party and not us.  The spec seems to say it's this way too:  https://openid.net/specs/openid-connect-core-1_0.html#IDToken

Comment (hjeeves) [reply cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/OIDCUtil.java:339]: I'll switch that back - somewhere I'd found one example that appeared to read differently, but I can't find it again. Ultimately as long as the value put in & tested for are the same, that's all that seems to matter.

File: cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/TokenAction.java [hunk 2]
```diff
@@ -111,12 +111,16 @@ public void doAction() throws Exception {
         // (our config makes clients post the secret: "token_endpoint_auth_methods_supported: client_secret_post")
         String clientSecret = syncInput.getParameter("client_secret");
         if (clientID == null || clientSecret == null) {
-            sendError("invalid_client");
+            sendError("invalid_client: missing id or secret");
```
Comment (brianmajor) [inline cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/TokenAction.java:114]: I'm pretty sure the error messages are standard and can't be modified to add extra info.
https://openid.net/specs/openid-connect-core-1_0.html#TokenErrorResponse

File: cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/TokenAction.java [hunk 3]
```diff
@@ -111,12 +111,16 @@ public void doAction() throws Exception {
...
             return;
         }
         RelyParty rp = OIDCUtil.getRelyParty(clientID);
-        if (rp == null || !rp.getClientSecret().equals(clientSecret)) {
-            sendError("invalid_client");
+        if (rp == null) {
+            sendError("invalid_client: reply party not found.");
```
Comment (brianmajor) [inline cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/TokenAction.java:119]: same thing here

File: cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/TokenAction.java [hunk 4]
```diff
@@ -111,12 +111,16 @@ public void doAction() throws Exception {
...
+            return;
+        }
+        if (!rp.getClientSecret().equals(clientSecret)) {
+            sendError("invalid_client: invalid secret");
```
Comment (brianmajor) [inline cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/TokenAction.java:123]: and here

**Review by brianmajor — CHANGES_REQUESTED** [review]
Just the error response changes.  I really think you've cleaned up the code well and simplified many areas.

**Review by brianmajor — APPROVED** [review]

**Pull request merged**